core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintService.java

```diff
@@ -176,11 +176,23 @@ public PlatformRecommendation getRecommendation(Long workspaceId, String bluepri
         return getAllAvailableInWorkspace(workspace);
     }
 
-    public Set<BlueprintView> getAllAvailableViewInWorkspace(Long workspaceId) {
+    public Set<BlueprintView> getAllAvailableViewInWorkspaceAndFilterBySdxReady(Long workspaceId, Boolean withSdx) {
         User user = getLoggedInUser();
         Workspace workspace = getWorkspaceService().get(workspaceId, user);
         updateDefaultBlueprintCollection(workspace);
-        return blueprintViewRepository.findAllByNotDeletedInWorkspace(workspaceId);
+        Set<BlueprintView> allByNotDeletedInWorkspace = blueprintViewRepository.findAllByNotDeletedInWorkspace(workspaceId);
+        if (withSdx) {
+            return allByNotDeletedInWorkspace;
+        }
+        return allByNotDeletedInWorkspace.stream().filter(it -> !isSDXReady(it)).collect(Collectors.toSet());
+    }
+
+    private boolean isSDXReady(BlueprintView blueprintView) {
+        if (blueprintView.getTags() == null || blueprintView.getTags().getValue() == null) {
+            return false;
+        }
+        Boolean sdxReady = blueprintView.getTags().getValue("shared_services_ready" );
```

Comment:
There is already a string constant for "shared_services_ready", I would use that.